modules/programs/git.nix

```diff
@@ -296,16 +296,71 @@ in {
           '';
         };
       };
+
+      diff-so-fancy = {
+        enable = mkEnableOption "" // {
+          description = ''
+            Enable the <command>diff-so-fancy</command> diff colorizer.
+            See <link xlink:href="https://github.com/so-fancy/diff-so-fancy" />.
+          '';
+        };
+        markEmptyLines = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = ''
+            Should the first block of an empty line be colored.
+          '';
+        };
+        changeHunkIndicators = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = ''
+            Simplify git header chunks to a more human readable format.
+          '';
+        };
+        stripLeadingSymbols = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = ''
+            Should the pesky + or - at line-start be removed.
+          '';
+        };
+        useUnicodeRuler = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = ''
+            By default, the separator for the file header uses Unicode
+            line-drawing characters. If this is causing output errors on
+            your terminal, set this to false to use ASCII characters instead.
+          '';
+        };
+        rulerWidth = mkOption {
+          type = types.nullOr types.int;
+          default = null;
+          example = false;
+          description = ''
+            By default, the separator for the file header spans the full
+            width of the terminal. Use this setting to set the width of
+            the file header manually.
+          '';
+        };
+      };
     };
   };
 
   config = mkIf cfg.enable (mkMerge [
     {
       home.packages = [ cfg.package ];
       assertions = [{
-        assertion = !(cfg.delta.enable && cfg.difftastic.enable);
+        assertion = !(cfg.delta.enable && cfg.difftastic.enable)
+          && !(cfg.delta.enable && cfg.diff-so-fancy.enable)
+          && !(cfg.difftastic.enable && cfg.diff-so-fancy.enable);
```

Comment:
You might just want to put them in a list, and count that at most one is enabled.

Reply:
@ambroisie Thanks for your review! I actually wanted to implement it that way, but couldn't figure out how to do it in Nix... Can you help with that?

Reply:
@arjan-s You could do this for example

```
assertion = let
  enabled = [
    cfg.delta.enable
    cfg.diff-so-fancy.enable
    cfg.difftastic.enable
  ];
in lib.length (lib.filter lib.id enabled) == 1;
```